### PR TITLE
Make all functions tibble friendly

### LINKED ISSUE #118: Make all functions tibble compatible
A user noted a bit ago that one of our functions broke when using a tibble (e.g., generated by read_csv with a PBDB url). At the time we just directed them to use read.csv instead to make a dataframe. However, it looks like there are some very small things we can do to make our functions compatible with both dataframes and tibbles: https://hugogruson.fr/posts/compa-tibble/

### PR DESCRIPTION
**Pull request by LewisAJones**
I've done a sweep through all our functions to address #118, adding `drop = TRUE` where appropriate. However, this was quite a lot to go through, so I'd appreciate an additional pair of eyes or two for reviewing! All tests are still passing so I haven't broken anything obvious but still, I think it is worthwhile having an additional check. And thanks for coming up with a simple solution in the issue!

Closes #118.

### CODE REVIEW

File: R/bin_time.R [hunk 1]
```diff
@@ -223,7 +228,7 @@ bin_time <- function(occdf, min_ma = "min_ma", max_ma = "max_ma",
 
       # Remove mid_ma for fossil occurrences (if not already present as input)
       if (rmcol == TRUE) {
-        occdf <- occdf[, -which(colnames(occdf) == "mid_ma")]
+        occdf <- occdf[, -which(colnames(occdf) == "mid_ma"), drop = TRUE]
```
Comment (willgearty) [inline R/bin_time.R:231]: I think this occdf will always have 2 or more columns (same as line 288), but it's probably fine to keep this?

File: R/tax_check.R [hunk 2]
```diff
@@ -245,11 +248,11 @@ tax_check <- function(taxdf, name = "genus", group = NULL, dis = 0.05,
...
   err$f1 <- as.vector(table(taxdf2[, "name"])[match(err$V1,
-                                                names(table(
-                                                  taxdf2[, "name"])))])
+                                                       names(table(
+                                                         taxdf2[, "name"])))])
   err$f2 <- as.vector(table(taxdf2[, "name"])[match(err$V2,
-                                                names(table(
-                                                  taxdf2[, "name"])))])
+                                                       names(table(
+                                                         taxdf2[, "name"])))])
```
Comment (willgearty) [inline R/tax_check.R:255]: Should these have drop = TRUE?

**Review by willgearty — APPROVED** [review]
I think I found a handful of other cases, but a third set of eyes would definitely be good. Otherwise, I just have some minor comments that can probably be ignored.

File: R/tax_check.R (hunk 2 above)
Comment (LewisAJones) [reply R/tax_check.R:255]: I think this is fine because taxdf2 is a `data.frame` generated in L181?

**Comment by LewisAJones** [comment]
@KEichenseer Do you think you will have chance to have a look over this PR within the next week or two? 🙏🏻

**Comment by LewisAJones** [comment]
> I think I found a handful of other cases, but a third set of eyes would definitely be good. Otherwise, I just have some minor comments that can probably be ignored.

**Comment by LewisAJones** [comment]
@KEichenseer Do you think you will have chance to have a look over this PR within the next week or two? 🙏🏻

**Comment by KEichenseer** [comment]
> @KEichenseer Do you think you will have chance to have a look over this PR within the next week or two? 🙏🏻

Sorry, I missed the previous comment, yes I'll give it a proper look this week.

**Comment by KEichenseer** [comment]
I think I messed something up when updating this PR from main - sorry! There were conflicts in `palaeorotate.R` and `test-palaeorotate.R` which I attempted to resolve by using the lines from main, but one `palaeorotate` test is now failing.

**Comment by LewisAJones** [comment]
Thanks @KEichenseer, there were quite a few major updates for `palaeorotate()` so I'm not surprised there were some conflicts. I believe I've resolved all of these now! Thanks for taking the time to look over things. Please have a final look when you can.

**Comment by KEichenseer** [comment]
> Thanks @KEichenseer, there were quite a few major updates for `palaeorotate()` so I'm not surprised there were some conflicts. I believe I've resolved all of these now! Thanks for taking the time to look over things. Please have a final look when you can.

Thanks @LewisAJones for fixing these issues.

I've made some minor remaining adjustments and approve merging.

**Review by KEichenseer — APPROVED** [review]
I tested all functions with tibbles and fixed a few remaining issues when using tibbles. 

I approve merging.

One remaining issue is that we haven't integrated comprehensive tests to check for tibble compatibility, but I don't know how to do that without duplicating all tests. So I think it is fine as is - we just need to watch out for tibble compatibility when adding / changing functions.

**Comment by LewisAJones** [comment]
Thanks @KEichenseer! I think this is good to go then. In the end, I feel it might have been easier just having an internal function that dealt with the conversion from tibbles to data frames and back 🥲 

Thanks to both you @willgearty and @KEichenseer for the review and contributions - it wasn't a very exciting PR but I suppose a useful one for tidyverse users!

**Comment by KEichenseer** [comment]
> Thanks @KEichenseer! I think this is good to go then. In the end, I feel it might have been easier just having an internal function that dealt with the conversion from tibbles to data frames and back 🥲
> 
> Thanks to both you @willgearty and @KEichenseer for the review and contributions - it wasn't a very exciting PR but I suppose a useful one for tidyverse users!

Easier for sure, on the bright side we are avoiding the `tibble` package dependency. Thanks for adding the bulk of these tedious `drop = TRUE` ;-)

**Pull request merged**